### DISCUSSION TITLE
How does gluetun decide which nordvpn server to connect to?

### DISCUSSION THREAD

**Discussion by ghost** [qdm12/gluetun#718] [Q&A] [answered]
I'm hoping that gluetun does some magic under the hood to determine the REGION server with the lowest load.  My question is, does it?

In the past I've written a bash script using [nordvpn-server-find](https://github.com/mrzool/nordvpn-server-find) that finds the server with the lowest utilization and copied its config to ./vpn/vpn.conf for a docker container to use.  Now that I'm switching to gluetun, I'm hoping such a script wouldn't be needed any more.

In the case that gluetun does connect to the recommended server, does it periodically recheck if another server has become recommended?

**Comment by qdm12** [comment, marked as answer]
No it does not, it picks at random a server from the list of filtered servers.

However that's an interesting optional feature, please create an issue linking your script.

Although it might get complicated for me to implement due to how the code is structured to be generic to support other providers. Maybe with Go 1.18 and generics (coming in like 3 months) that would be possible.

**Comment by ghost** [comment]
I'll update the script to change the SERVER_HOSTNAME in docker-compose.yml and create an issue after testing it thoroughly. Even if it can't be included in the container itself, it could still help others if added to the wiki.
